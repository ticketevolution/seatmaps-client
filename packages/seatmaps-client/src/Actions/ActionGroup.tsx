import React from "react";

export default function (props: React.HTMLProps<HTMLDivElement>) {
  return (
    <div
      {...props}
      style={{
        pointerEvents: "all",
        border: "2px solid lightgray",
        borderRadius: 5,
        display: "flex",
        backgroundColor: "white",
        ...(props.style || {}),
      }}
    />
  );
}
