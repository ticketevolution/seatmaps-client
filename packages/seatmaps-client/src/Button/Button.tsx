import React from "react";

interface Props extends React.HTMLProps<HTMLDivElement> {
  icon?: React.ReactNode;
  text?: string;
  isMobile?: boolean;
}

export default function Button({
  icon,
  text,
  style,
  isMobile,
  ...props
}: Props) {
  return (
    <div
      {...props}
      style={{
        backgroundColor: "transparent",
        borderRadius: 0,
        padding: isMobile ? "0.25em 0.5em" : "0.5em 1em",
        outline: "none",
        cursor: "pointer",
        border: "none",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        gap: isMobile ? "4px" : "6px",
        ...(style || {}),
      }}
    >
      {icon}
      {text}
    </div>
  );
}
