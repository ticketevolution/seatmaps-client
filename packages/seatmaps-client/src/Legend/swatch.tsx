import React, { CSSProperties } from "react";

export interface Props {
  style?: CSSProperties;
  color: string;
}

export default (props: Props) => (
  <div
    style={{
      ...(props.style || {}),
      height: 14,
      width: 14,
      borderRadius: 4,
      backgroundColor: props.color,
      display: "inline-block",
    }}
  />
);
