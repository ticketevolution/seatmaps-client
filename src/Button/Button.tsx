import React from 'react'

export default function Button (props: React.ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <button
      {...props}
      style={{
        background: 'white',
        color: '#4A4A4A',
        border: '1px solid #E0E0E0',
        borderRadius: 8,
        textDecoration: 'none',
        cursor: 'pointer',
        lineHeight: 1.5,
        fontSize: 16,
        ...props.style
      }}
    />
  )
}
