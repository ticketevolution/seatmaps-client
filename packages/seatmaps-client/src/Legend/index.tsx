import React, { Component } from "react";
import Button from "../Button";
import Swatch from "./swatch";
import { IconChevronDown } from "../icons/ChevronDown";
import { IconChevronUp } from "../icons/ChevronUp";

const formatCurrency = new Intl.NumberFormat(undefined, {
  style: "currency",
  currency: "USD",
  maximumFractionDigits: 0,
  minimumFractionDigits: 0,
}).format;

export interface Range {
  color: string;
  min: number;
  max: number;
}

export interface Props {
  ranges: Range[];
  isMobile?: boolean;
}

interface State {
  isOpen: boolean;
}

export default class Legend extends Component<Props, State> {
  static defaultProps = {
    isMobile: false,
  };

  state = {
    isOpen: false,
  };

  render() {
    const { isOpen } = this.state;
    const { ranges, isMobile } = this.props;

    return (
      <div style={{ position: "relative" }}>
        <Button
          onClick={() => this.setState({ isOpen: !isOpen })}
          icon={isOpen ? <IconChevronUp /> : <IconChevronDown />}
          text={`${isOpen ? "Hide " : "Show "}Map Legend`}
          isMobile={isMobile}
        />
        {ranges.length > 0 && isOpen && (
          <div
            style={{
              position: "absolute",
              backgroundColor: "white",
              right: -2,
              border: "2px solid lightgray",
              borderRadius: "0 0 5px 5px",
            }}
          >
            {ranges.map((range) => (
              <div key={range.color} style={{ padding: 8, textAlign: "left" }}>
                <Swatch color={range.color} style={{ marginRight: 8 }} />
                <span>
                  {formatCurrency(Math.floor(range.min))}
                  {" - "}
                  {formatCurrency(Math.ceil(range.max))}
                </span>
              </div>
            ))}
          </div>
        )}
      </div>
    );
  }
}
