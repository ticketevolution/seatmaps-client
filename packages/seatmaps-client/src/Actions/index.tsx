import React from "react";
import Legend from "../Legend";
import Button from "../Button";
import ActionGroup from "./ActionGroup";
import type { CostRange } from "../TicketMap";
import { IconPlus } from "../icons/Plus";
import { IconMinus } from "../icons/Minus";
import { IconUndo } from "../icons/Undo";
import { IconClose } from "../icons/Close";

export interface Props {
  onClearSelection?: () => void;
  showLegend?: boolean;
  showControls?: boolean;
  ranges: CostRange[];
  onZoomIn: () => void;
  onZoomOut: () => void;
  onResetZoom: () => void;
}

interface DefaultProps {
  showLegend: boolean;
  showControls: boolean;
  onClearSelection(): void;
}

interface State {
  isMobile: boolean;
}

export default class Actions extends React.Component<
  Props & DefaultProps,
  State
> {
  container = React.createRef<HTMLDivElement>();
  timer?: number;

  state: State = {
    // initial value, would correct itself after first render
    // TODO: we should refactor this to an FC component and some variant of useMediaQuery
    isMobile: false,
  };

  static defaultProps: DefaultProps = {
    showLegend: true,
    showControls: true,
    onClearSelection: () => {},
  };

  get styles(): { [key: string]: React.CSSProperties } {
    return {
      container: {
        position: "absolute",
        top: this.state.isMobile ? 0 : 10,
        left: this.state.isMobile ? 0 : 10,
        right: this.state.isMobile ? 0 : 10,
        bottom: this.state.isMobile ? 0 : 10,
        display: "flex",
        alignItems: "flex-start",
        justifyContent: "space-between",
        opacity: 0.9,
        pointerEvents: "none",
        whiteSpace: "nowrap",
        maxWidth: "100%",
        fontSize: "0.75em",
      },
      icon: {
        display: "inline-block",
        fontSize: "inherit",
        height: "1.333333333em",
        overflow: "visible",
        verticalAlign: "-0.125em",
      },
    };
  }

  componentDidMount() {
    this.timer = window.setInterval(this.updateIsMobile, 200);
  }

  componentWillUnmount() {
    window.clearInterval(this.timer);
  }

  updateIsMobile = () => {
    const currentContainer = this.getCurrentContainer();
    if (!currentContainer) {
      return;
    }

    const isMobile =
      window.innerWidth < 520 || currentContainer.clientWidth < 400;

    if (this.state.isMobile !== isMobile) {
      this.setState({ isMobile });
    }
  };

  getCurrentContainer = () => {
    return this.container.current;
  };

  render() {
    const { isMobile } = this.state;
    const { showControls, showLegend } = this.props;

    const showLeftActions = showControls || (isMobile && showLegend);
    const showRightActions = !isMobile && showLegend;

    return (
      <div
        style={this.styles.container}
        ref={this.container}
        data-testid="seatmaps-actions-menu"
      >
        {showLeftActions && (
          <ActionGroup>
            {!isMobile && showControls && (
              <React.Fragment>
                <Button
                  data-rh="Default"
                  data-custom-at="right"
                  onClick={this.props.onZoomIn}
                  icon={<IconPlus />}
                  isMobile={isMobile}
                  style={{ borderRight: "2px solid lightgray" }}
                  name="zoom-in"
                  data-testid="zoom-in"
                />

                <Button
                  onClick={this.props.onZoomOut}
                  icon={<IconMinus />}
                  isMobile={isMobile}
                  style={{ borderRight: "2px solid lightgray" }}
                  name="zoom-out"
                  data-testid="zoom-out"
                />

                <Button
                  onClick={this.props.onResetZoom}
                  icon={<IconUndo />}
                  text="Reset Zoom"
                  isMobile={isMobile}
                  style={{ borderRight: "2px solid lightgray" }}
                  name="reset-zoom"
                  data-testid="reset-zoom"
                />
              </React.Fragment>
            )}
            {showControls && (
              <Button
                className="clear-selection"
                onClick={this.props.onClearSelection}
                icon={<IconClose />}
                text={`Clear${isMobile ? "" : " All"}`}
                isMobile={isMobile}
                style={{
                  borderRight: isMobile ? "2px solid lightgray" : undefined,
                }}
                name="clear-selection"
              />
            )}
            {isMobile && showLegend && (
              <Legend isMobile ranges={this.props.ranges} />
            )}
          </ActionGroup>
        )}
        {showRightActions && (
          <ActionGroup>
            <Legend ranges={this.props.ranges} />
          </ActionGroup>
        )}
      </div>
    );
  }
}
