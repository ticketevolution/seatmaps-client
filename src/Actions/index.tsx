import React from 'react'
import svgPanZoom from 'svg-pan-zoom/src/svg-pan-zoom.js'
import { faTimesCircle, faPlus, faMinus, faUndoAlt } from '@fortawesome/free-solid-svg-icons'
import Legend from '../Legend'
import Button from '../Button'
import ActionGroup from './ActionGroup'
import { CostRange } from 'src/TicketMap'

interface Props {
  mapSvg: SVGSVGElement
  onClearSelection(): void
  showLegend: boolean
  ranges: CostRange[]
}

interface State {
  isMobile: boolean
}

export default class Actions extends React.Component<Props, State> {
  container?: HTMLDivElement | null
  mapZoom: any // eslint-disable-line @typescript-eslint/no-explicit-any
  timer?: number

  constructor (props: Props) {
    super(props)

    this.mapZoom = svgPanZoom(this.props.mapSvg, {
      minZoom: 0.8
    })
  }

  get styles (): { [key: string]: React.CSSProperties } {
    return {
      container: {
        position: 'absolute',
        top: this.state.isMobile ? 0 : 10,
        left: this.state.isMobile ? 0 : 10,
        right: this.state.isMobile ? 0 : 10,
        bottom: this.state.isMobile ? 0 : 10,
        display: 'flex',
        alignItems: 'flex-start',
        justifyContent: 'space-between',
        opacity: 0.9,
        pointerEvents: 'none',
        whiteSpace: 'nowrap',
        maxWidth: '100%',
        fontSize: '0.75em'
      },
      icon: {
        display: 'inline-block',
        fontSize: 'inherit',
        height: '1.333333333em',
        overflow: 'visible',
        verticalAlign: '-0.125em'
      }
    }
  }

  componentDidMount () {
    this.timer = window.setInterval(this.updateIsMobile, 200)
  }

  componentWillUnmount () {
    window.clearInterval(this.timer)
  }

  updateIsMobile = () => {
    if (!this.container) {
      return
    }

    const isMobile = this.container.clientWidth < 700

    if (this.state.isMobile !== isMobile) {
      this.setState({ isMobile })
    }
  }

  render () {
    const { isMobile } = this.state

    return (
      <div style={this.styles.container} ref={ref => { this.container = ref }}>
        <ActionGroup>
          <Button
            data-rh='Default'
            data-custom-at='right'
            onClick={() => this.mapZoom.zoomIn()}
            icon={faPlus}
            isMobile={isMobile}
            style={{ borderRight: '2px solid lightgray' }}
          />

          <Button
            onClick={() => this.mapZoom.zoomOut()}
            icon={faMinus}
            isMobile={isMobile}
            style={{ borderRight: '2px solid lightgray' }}
          />

          <Button
            onClick={() => this.mapZoom.reset()}
            icon={faUndoAlt}
            text='Reset Zoom'
            isMobile={isMobile}
            style={{ borderRight: '2px solid lightgray' }}
          />

          <Button
            onClick={() => this.props.onClearSelection()}
            icon={faTimesCircle}
            text={`Clear${isMobile ? '' : ' All'}`}
            isMobile={isMobile}
            style={{ borderRight: isMobile ? '2px solid lightgray' : undefined }}
          />

          {isMobile && this.props.showLegend && <Legend isMobile ranges={this.props.ranges} />}
        </ActionGroup>
        {!isMobile && <ActionGroup>
          {this.props.showLegend && <Legend ranges={this.props.ranges} />}
        </ActionGroup>}
      </div>
    )
  }
}
