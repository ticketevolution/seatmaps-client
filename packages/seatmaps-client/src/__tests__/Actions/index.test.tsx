import "jest-enzyme";

import React from "react";
import { act } from "react-dom/test-utils";
import { mount, ReactWrapper } from "enzyme";
import Actions, { Props } from "../../Actions";
import ActionGroup from "../../Actions/ActionGroup";
import Legend from "../../Legend";
import Button from "../../Button";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";

describe("Actions", () => {
  let wrapper: ReactWrapper<Actions["props"], Actions["state"], Actions>;
  let props: Props;

  beforeEach(() => {
    props = {
      ranges: [],
      onClearSelection: jest.fn(),
      onZoomIn: jest.fn(),
      onZoomOut: jest.fn(),
      onResetZoom: jest.fn(),
    };
    wrapper = mount<Actions>(<Actions {...props} />);
  });

  it("renders", () => {
    expect(wrapper.isEmptyRender()).toEqual(false);
  });

  it("renders the actions and legend in one group on mobile browsers", () => {
    wrapper.setProps({ showControls: true, showLegend: true });
    act(() => {
      wrapper.setState({ isMobile: true });
    });
    expect(wrapper.find(ActionGroup)).toHaveLength(1);
  });

  it("renders the actions and legend in separate groups on mobile browsers", () => {
    wrapper.setProps({ showControls: true, showLegend: true });
    act(() => {
      wrapper.setState({ isMobile: false });
    });
    expect(wrapper.find(ActionGroup)).toHaveLength(2);
    expect(wrapper.find(ActionGroup).at(1).find(Legend)).toHaveLength(1);
  });

  describe("when controls are visible", () => {
    beforeEach(() => {
      wrapper.setProps({ showControls: true });
    });

    describe("on desktop browsers", () => {
      beforeEach(() => {
        act(() => {
          wrapper.setState({ isMobile: false });
        });
      });

      it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
      });

      it("renders a clear selection button", () => {
        expect(wrapper.exists('[name="clear-selection"]')).toEqual(true);
      });

      it("renders a zoom in button", () => {
        expect(wrapper.exists('[name="zoom-in"]')).toEqual(true);
      });

      it("renders a zoom out button", () => {
        expect(wrapper.exists('[name="zoom-out"]')).toEqual(true);
      });

      it("renders a reset zoom button", () => {
        expect(wrapper.exists('[name="reset-zoom"]')).toEqual(true);
      });

      it("renders a button that zooms in", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onZoomIn).toHaveBeenCalled();
      });

      it("renders a button that zooms out", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onZoomOut).toHaveBeenCalled();
      });

      it("renders a button that resets zoom", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onResetZoom).toHaveBeenCalled();
      });

      it("renders a button that clears selections", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onClearSelection).toHaveBeenCalled();
      });
    });

    describe("on mobile browsers", () => {
      beforeEach(() => {
        act(() => {
          wrapper.instance().setState({ isMobile: true });
        });
      });

      it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
      });

      it("renders a clear selection button", () => {
        expect(wrapper.exists('[name="clear-selection"]')).toEqual(true);
      });

      it("does not render a button that zooms in", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onZoomIn).not.toHaveBeenCalled();
      });

      it("does not render a button that zooms out", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onZoomOut).not.toHaveBeenCalled();
      });

      it("does not render a button that resets zoom", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onResetZoom).not.toHaveBeenCalled();
      });

      it("renders a button that clears selections", () => {
        wrapper.find(Button).forEach((button) => button.simulate("click"));
        expect(props.onClearSelection).toHaveBeenCalled();
      });
    });

    describe("when component is unmounted", () => {
      beforeEach(() => {
        jest.spyOn(window, "clearInterval");
      });

      it("should clear the interval", () => {
        wrapper.unmount();
        expect(clearInterval).toHaveBeenCalled();
      });
    });

    describe("updateIsMobile", () => {
      it("should not call setState if the container is not defined", () => {
        const getCurrentContainerMock =
          jest.fn<Actions["getCurrentContainer"]>();
        getCurrentContainerMock.mockReturnValue(null);
        wrapper.instance().getCurrentContainer = getCurrentContainerMock;
        jest.spyOn(wrapper.instance(), "setState");
        wrapper.update();

        wrapper.instance().updateIsMobile();

        expect(getCurrentContainerMock).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().setState).not.toHaveBeenCalled();
      });

      it("should set isMobile true if container width is mobile", () => {
        const getCurrentContainerMock =
          jest.fn<Actions["getCurrentContainer"]>();
        getCurrentContainerMock.mockReturnValue({
          clientWidth: 399,
        } as unknown as HTMLDivElement);
        wrapper.instance().getCurrentContainer = getCurrentContainerMock;
        act(() => {
          wrapper.instance().setState({ isMobile: false });
        });
        jest.spyOn(wrapper.instance(), "setState");
        wrapper.update();

        act(() => {
          wrapper.instance().updateIsMobile();
        });

        expect(getCurrentContainerMock).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().setState).toHaveBeenCalledWith({
          isMobile: true,
        });
      });

      it("should set isMobile true if window width is mobile", () => {
        const getCurrentContainerMock =
          jest.fn<Actions["getCurrentContainer"]>();
        getCurrentContainerMock.mockReturnValue({
          clientWidth: 400,
        } as unknown as HTMLDivElement);
        Object.defineProperty(window, "innerWidth", {
          writable: true,
          configurable: true,
          value: 519,
        });
        wrapper.instance().getCurrentContainer = getCurrentContainerMock;
        act(() => {
          wrapper.instance().setState({ isMobile: false });
        });
        jest.spyOn(wrapper.instance(), "setState");
        wrapper.update();

        act(() => {
          wrapper.instance().updateIsMobile();
        });

        expect(getCurrentContainerMock).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().setState).toHaveBeenCalledWith({
          isMobile: true,
        });
      });
    });
  });
});
