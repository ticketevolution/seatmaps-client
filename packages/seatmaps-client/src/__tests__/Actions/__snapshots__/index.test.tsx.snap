// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions when controls are visible on desktop browsers matches snapshot 1`] = `
<Actions
  onClearSelection={[MockFunction]}
  onResetZoom={[MockFunction]}
  onZoomIn={[MockFunction]}
  onZoomOut={[MockFunction]}
  ranges={[]}
  showControls={true}
  showLegend={true}
>
  <div
    data-testid="seatmaps-actions-menu"
    style={
      {
        "alignItems": "flex-start",
        "bottom": 10,
        "display": "flex",
        "fontSize": "0.75em",
        "justifyContent": "space-between",
        "left": 10,
        "maxWidth": "100%",
        "opacity": 0.9,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 10,
        "top": 10,
        "whiteSpace": "nowrap",
      }
    }
  >
    <_default>
      <div
        style={
          {
            "backgroundColor": "white",
            "border": "2px solid lightgray",
            "borderRadius": 5,
            "display": "flex",
            "pointerEvents": "all",
          }
        }
      >
        <Button
          data-custom-at="right"
          data-rh="Default"
          data-testid="zoom-in"
          icon={<IconPlus />}
          isMobile={false}
          name="zoom-in"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
        >
          <div
            data-custom-at="right"
            data-rh="Default"
            data-testid="zoom-in"
            name="zoom-in"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconPlus>
              <svg
                height="16"
                viewBox="0 0 448 512"
                width="14"
              >
                <path
                  d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32v144H48c-17.7 0-32 14.3-32 32s14.3 32 32 32h144v144c0 17.7 14.3 32 32 32s32-14.3 32-32V288h144c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                  fill="currentColor"
                />
              </svg>
            </IconPlus>
          </div>
        </Button>
        <Button
          data-testid="zoom-out"
          icon={<IconMinus />}
          isMobile={false}
          name="zoom-out"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
        >
          <div
            data-testid="zoom-out"
            name="zoom-out"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconMinus>
              <svg
                height="16"
                viewBox="0 0 448 512"
                width="14"
              >
                <path
                  d="M432 256c0 17.7-14.3 32-32 32H48c-17.7 0-32-14.3-32-32s14.3-32 32-32h352c17.7 0 32 14.3 32 32z"
                  fill="currentColor"
                />
              </svg>
            </IconMinus>
          </div>
        </Button>
        <Button
          data-testid="reset-zoom"
          icon={<IconUndo />}
          isMobile={false}
          name="reset-zoom"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
          text="Reset Zoom"
        >
          <div
            data-testid="reset-zoom"
            name="reset-zoom"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconUndo>
              <svg
                height="16"
                viewBox="0 0 512 512"
                width="16"
              >
                <path
                  d="M48.5 224H40c-13.3 0-24-10.7-24-24V72c0-9.7 5.8-18.5 14.8-22.2S50.1 48.1 57 55l41.6 41.6c87.6-86.5 228.7-86.2 315.8 1 87.5 87.5 87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3c-62.2-62.2-162.7-62.5-225.3-1L185 183c6.9 6.9 8.9 17.2 5.2 26.2S177.7 224 168 224H48.5z"
                  fill="currentColor"
                />
              </svg>
            </IconUndo>
            Reset Zoom
          </div>
        </Button>
        <Button
          className="clear-selection"
          icon={<IconClose />}
          isMobile={false}
          name="clear-selection"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": undefined,
            }
          }
          text="Clear All"
        >
          <div
            className="clear-selection"
            name="clear-selection"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": undefined,
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconClose>
              <svg
                height="16"
                viewBox="0 0 512 512"
                width="16"
              >
                <path
                  d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-81-337c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                  fill="currentColor"
                />
              </svg>
            </IconClose>
            Clear All
          </div>
        </Button>
      </div>
    </_default>
    <_default>
      <div
        style={
          {
            "backgroundColor": "white",
            "border": "2px solid lightgray",
            "borderRadius": 5,
            "display": "flex",
            "pointerEvents": "all",
          }
        }
      >
        <Legend
          isMobile={false}
          ranges={[]}
        >
          <div
            style={
              {
                "position": "relative",
              }
            }
          >
            <Button
              icon={<IconChevronDown />}
              isMobile={false}
              onClick={[Function]}
              text="Show Map Legend"
            >
              <div
                onClick={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "border": "none",
                    "borderRadius": 0,
                    "cursor": "pointer",
                    "display": "flex",
                    "gap": "6px",
                    "justifyContent": "center",
                    "outline": "none",
                    "padding": "0.5em 1em",
                  }
                }
              >
                <IconChevronDown>
                  <svg
                    height="16"
                    viewBox="0 0 512 512"
                    width="16"
                  >
                    <path
                      d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                      fill="currentColor"
                    />
                  </svg>
                </IconChevronDown>
                Show Map Legend
              </div>
            </Button>
          </div>
        </Legend>
      </div>
    </_default>
  </div>
</Actions>
`;

exports[`Actions when controls are visible on mobile browsers matches snapshot 1`] = `
<Actions
  onClearSelection={[MockFunction]}
  onResetZoom={[MockFunction]}
  onZoomIn={[MockFunction]}
  onZoomOut={[MockFunction]}
  ranges={[]}
  showControls={true}
  showLegend={true}
>
  <div
    data-testid="seatmaps-actions-menu"
    style={
      {
        "alignItems": "flex-start",
        "bottom": 10,
        "display": "flex",
        "fontSize": "0.75em",
        "justifyContent": "space-between",
        "left": 10,
        "maxWidth": "100%",
        "opacity": 0.9,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 10,
        "top": 10,
        "whiteSpace": "nowrap",
      }
    }
  >
    <_default>
      <div
        style={
          {
            "backgroundColor": "white",
            "border": "2px solid lightgray",
            "borderRadius": 5,
            "display": "flex",
            "pointerEvents": "all",
          }
        }
      >
        <Button
          data-custom-at="right"
          data-rh="Default"
          data-testid="zoom-in"
          icon={<IconPlus />}
          isMobile={false}
          name="zoom-in"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
        >
          <div
            data-custom-at="right"
            data-rh="Default"
            data-testid="zoom-in"
            name="zoom-in"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconPlus>
              <svg
                height="16"
                viewBox="0 0 448 512"
                width="14"
              >
                <path
                  d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32v144H48c-17.7 0-32 14.3-32 32s14.3 32 32 32h144v144c0 17.7 14.3 32 32 32s32-14.3 32-32V288h144c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                  fill="currentColor"
                />
              </svg>
            </IconPlus>
          </div>
        </Button>
        <Button
          data-testid="zoom-out"
          icon={<IconMinus />}
          isMobile={false}
          name="zoom-out"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
        >
          <div
            data-testid="zoom-out"
            name="zoom-out"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconMinus>
              <svg
                height="16"
                viewBox="0 0 448 512"
                width="14"
              >
                <path
                  d="M432 256c0 17.7-14.3 32-32 32H48c-17.7 0-32-14.3-32-32s14.3-32 32-32h352c17.7 0 32 14.3 32 32z"
                  fill="currentColor"
                />
              </svg>
            </IconMinus>
          </div>
        </Button>
        <Button
          data-testid="reset-zoom"
          icon={<IconUndo />}
          isMobile={false}
          name="reset-zoom"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": "2px solid lightgray",
            }
          }
          text="Reset Zoom"
        >
          <div
            data-testid="reset-zoom"
            name="reset-zoom"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": "2px solid lightgray",
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconUndo>
              <svg
                height="16"
                viewBox="0 0 512 512"
                width="16"
              >
                <path
                  d="M48.5 224H40c-13.3 0-24-10.7-24-24V72c0-9.7 5.8-18.5 14.8-22.2S50.1 48.1 57 55l41.6 41.6c87.6-86.5 228.7-86.2 315.8 1 87.5 87.5 87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3c-62.2-62.2-162.7-62.5-225.3-1L185 183c6.9 6.9 8.9 17.2 5.2 26.2S177.7 224 168 224H48.5z"
                  fill="currentColor"
                />
              </svg>
            </IconUndo>
            Reset Zoom
          </div>
        </Button>
        <Button
          className="clear-selection"
          icon={<IconClose />}
          isMobile={false}
          name="clear-selection"
          onClick={[MockFunction]}
          style={
            {
              "borderRight": undefined,
            }
          }
          text="Clear All"
        >
          <div
            className="clear-selection"
            name="clear-selection"
            onClick={[MockFunction]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "borderRadius": 0,
                "borderRight": undefined,
                "cursor": "pointer",
                "display": "flex",
                "gap": "6px",
                "justifyContent": "center",
                "outline": "none",
                "padding": "0.5em 1em",
              }
            }
          >
            <IconClose>
              <svg
                height="16"
                viewBox="0 0 512 512"
                width="16"
              >
                <path
                  d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-81-337c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                  fill="currentColor"
                />
              </svg>
            </IconClose>
            Clear All
          </div>
        </Button>
      </div>
    </_default>
    <_default>
      <div
        style={
          {
            "backgroundColor": "white",
            "border": "2px solid lightgray",
            "borderRadius": 5,
            "display": "flex",
            "pointerEvents": "all",
          }
        }
      >
        <Legend
          isMobile={false}
          ranges={[]}
        >
          <div
            style={
              {
                "position": "relative",
              }
            }
          >
            <Button
              icon={<IconChevronDown />}
              isMobile={false}
              onClick={[Function]}
              text="Show Map Legend"
            >
              <div
                onClick={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "border": "none",
                    "borderRadius": 0,
                    "cursor": "pointer",
                    "display": "flex",
                    "gap": "6px",
                    "justifyContent": "center",
                    "outline": "none",
                    "padding": "0.5em 1em",
                  }
                }
              >
                <IconChevronDown>
                  <svg
                    height="16"
                    viewBox="0 0 512 512"
                    width="16"
                  >
                    <path
                      d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                      fill="currentColor"
                    />
                  </svg>
                </IconChevronDown>
                Show Map Legend
              </div>
            </Button>
          </div>
        </Legend>
      </div>
    </_default>
  </div>
</Actions>
`;
